baked_in.go

```diff
@@ -767,6 +769,11 @@ func fieldExcludes(fl FieldLevel) bool {
 	return !strings.Contains(field.String(), currentField.String())
 }
 
+func regex(fl FieldLevel) bool {
+	regex := regexp.MustCompile(fl.Param())
```

Comment:
Running `MustCompile` on every validation is expensive (and an anti-pattern). Is there a way to call compile once and cache the regexp?